notebook/services/api/handlers.py

```diff
@@ -18,6 +18,13 @@ def get(self):
         self.set_header('Content-Type', 'text/x-yaml')
         return web.StaticFileHandler.get(self, 'api.yaml')
 
+    @web.authenticated
+    def options(self, section_name):
```

Comment:
I think you will need to accept `*args, **kwargs`  to deal with tornado's URL parsing